### ISSUE TITLE
"Argument type partially unknown" in calls to `str.translate` and `str.maketrans`

### ISSUE THREAD

**Issue by ottumm** [microsoft/pyright#5635]
**Describe the bug**
Pyright `1.1.319` errors on the following code, with:

```
  test.py - error: Argument type is partially unknown
    Argument corresponds to parameter "__x" in function "maketrans"
    Argument type is "dict[str, Unknown]" (reportUnknownArgumentType)
  test.py - error: Argument type is partially unknown
    Argument corresponds to parameter "__table" in function "translate"
    Argument type is "dict[int, Unknown]" (reportUnknownArgumentType)
```

**Code**
```python
def foo(s: str) -> str:
    return s.translate(
        str.maketrans(
            {
                "\\": "\\\\",
                '"': '\\"',
            }
        )
    )
```

**VS Code extension or command-line**
This error reproduces when running pyright from the command line with version `1.1.319`. Version `1.1.318` does not exhibit this error.

**Comment by erictraut** [comment]
Thanks for the bug report. 

Here's a simplified example:

```python
# pyright: strict
from typing import Protocol, TypeVar

class ProtoT(Protocol):
    def __getitem__(self, __key: int) -> str | int | None:
        ...

_T = TypeVar("_T")

def func1(__x: dict[str, _T]) -> dict[int, _T]:
    ...

x1: ProtoT = func1({"": ""})
```

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.321, which I just published. It will also be included in a future release of pylance.
